src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -720,6 +739,38 @@ private boolean isEmptyCatch(DetailAST ast, int parentType) {
         return allowEmptyCatches && isEmptyBlock(ast, parentType, TokenTypes.LITERAL_CATCH);
     }
 
+    /**
+     * Tests if the given {@code DetailAst} is part of an allowed empty
+     * switch case or default block.
+     *
+     * @param ast the {@code DetailAst} to test.
+     * @return {@code true} if {@code ast} makes up part of an allowed
+     *         empty switch case or default block.
+     */
+    private boolean isEmptySwitchBlockStatement(DetailAST ast) {
+        boolean isEmptySwitchBlockStatement = false;
+
+        if (allowEmptySwitchBlockStatements) {
+            final DetailAST parent = ast.getParent();
+            final DetailAST grandParent = parent.getParent();
+
+            final boolean isEmptyCaseInSwitchRule =
+                    isEmptyBlock(ast, parent.getType(), TokenTypes.SWITCH_RULE);
+
+            final boolean isEmptyCaseGroupCheckedFromLcurly = isEmptyBlock(ast,
+                    grandParent.getType(), TokenTypes.CASE_GROUP);
+
+            final boolean isEmptyCaseGroupCheckFromRcurly =
+                    parent.getFirstChild().getType() == TokenTypes.RCURLY
+                      && grandParent.getParent().getType() == TokenTypes.CASE_GROUP;
```

Comment:
Help me to understand what these are, it is not obvious why we should care about from lcurly, rcurly.

Reply:
The default implementation of `isEmptyBlock()` [when we have a `RCURLY` is to check that the grandparent is equal to the token we want to match with it](https://github.com/checkstyle/checkstyle/blob/f187bc86640851d6bde1dc81a8fce49d0069dff1/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java#L642-L644) (which is `CASE_GROUP` in this example). 


        |       |   |--CASE_GROUP -> CASE_GROUP [5:12]
        |       |   |   |--LITERAL_CASE -> case [5:12]
        |       |   |   |   |--EXPR -> EXPR [5:17]
        |       |   |   |   |   `--NUM_INT -> 4 [5:17]
        |       |   |   |   `--COLON -> : [5:18]
        |       |   |   `--SLIST -> SLIST [5:20]         <------- grandparent but we need to match with case groupt
        |       |   |       `--SLIST -> { [5:20]
        |       |   |           `--RCURLY -> } [5:21]


So I had to make a custom implementation for this special case and check the parent of the grand parent   `grandParent.getParent().getType() == TokenTypes.CASE_GROUP;`



In the `SLIST` part [we just check that the second and the third arguments are equal](https://github.com/checkstyle/checkstyle/blob/f187bc86640851d6bde1dc81a8fce49d0069dff1/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java#L647-L649) and [we are sure that this will be true because I am passing the grandparent as the second parameter](https://github.com/checkstyle/checkstyle/blob/f187bc86640851d6bde1dc81a8fce49d0069dff1/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java#L761)


In the case of `->` syntax, we don't have these exceptions because the grandparent of rcurly is `SWITCH_RULE` not that extra `SLIST`

```
        |       |   |--SWITCH_RULE -> SWITCH_RULE [8:12]
        |       |   |   |--LITERAL_CASE -> case [8:12]
        |       |   |   |   `--EXPR -> EXPR [8:17]
        |       |   |   |       `--NUM_INT -> 3 [8:17]
        |       |   |   |--LAMBDA -> -> [8:19]
        |       |   |   `--SLIST -> { [8:22]
        |       |   |       `--RCURLY -> } [8:23]

```

Reply:
Thanks for the explanation, I was hoping to think of a better name here, but nothing comes to mind 😂let’s roll with it